calculator/models.py

```diff
@@ -73,29 +73,28 @@ class Loan(models.Model):
         validators=[MinValueValidator(Decimal("0.01"))],
     )
 
-    def _instalment_adjustment(self):
+    def _rate_adjustment(self):
```

Comment:
I think the new value rate should be saved on the database, maybe using the method save like installment.
Something like this:

def save(self, *args, **kwargs):
        self.rate = self._rate_adjustment()
        self.instalment += self.calculate_instalment()
        super(Loan, self).save(*args, **kwargs)

And so you can remove the call method _rate_adjustment from calculate_instalment.

Reply:
I tried to keep both independent, but since we need the adjust value to find instalment value, its not possible to separate them. 
And I implemented your suggestion about tracking the auto adjust in the database, so I created a field for it.